js/apps/admin-ui/src/user/UserForm.tsx

```diff
@@ -133,284 +137,217 @@ export const UserForm = ({
-              )}
+              label={t("emailVerified")}
+              labelIcon={t("emailVerifiedHelp")}
             />
-          </FormGroup>
-          <UserProfileFields
-            form={form}
-            userProfileMetadata={userProfileMetadata}
-            hideReadOnly={!user}
-            supportedLocales={realm.supportedLocales || []}
-            t={
-              ((key: unknown, params) =>
-                t(key as string, params as any)) as TFunction
-            }
-          />
-        </>
-      ) : (
-        <>
-          {!realm.registrationEmailAsUsername && (
-            <FormGroup
-              label={t("username")}
-              fieldId="kc-username"
-              isRequired
-              validated={errors.username ? "error" : "default"}
-              helperTextInvalid={t("required")}
-            >
-              <KeycloakTextInput
-                id="kc-username"
-                isReadOnly={
+            <UserProfileFields
+              form={form}
+              userProfileMetadata={userProfileMetadata}
+              hideReadOnly={!user}
+              supportedLocales={realm.supportedLocales || []}
+              t={
+                ((key: unknown, params) =>
+                  t(key as string, params as any)) as TFunction
+              }
+            />
+          </>
+        ) : (
+          <>
+            {!realm.registrationEmailAsUsername && (
+              <TextControl
+                name="username"
+                label={t("username")}
+                readOnlyVariant={
                   !!user?.id &&
                   !realm.editUsernameAllowed &&
                   realm.editUsernameAllowed !== undefined
+                    ? "default"
+                    : undefined
                 }
-                {...register("username")}
+                rules={{
+                  required: t("required"),
+                }}
               />
-            </FormGroup>
-          )}
-          <FormGroup
-            label={t("email")}
-            fieldId="kc-email"
-            validated={errors.email ? "error" : "default"}
-            helperTextInvalid={t("emailInvalid")}
-          >
-            <KeycloakTextInput
+            )}
+            <TextControl
+              name="email"
+              label={t("email")}
               type="email"
-              id="kc-email"
-              data-testid="email-input"
-              {...register("email", {
-                pattern: emailRegexPattern,
-              })}
+              rules={{
+                pattern: {
+                  value: emailRegexPattern,
+                  message: t("emailInvalid"),
+                },
+              }}
             />
-          </FormGroup>
+            <SwitchControl
+              name="emailVerified"
+              label={t("emailVerified")}
+              labelIcon={t("emailVerifiedHelp")}
+              labelOn={t("yes")}
+              labelOff={t("no")}
+            />
```

Comment:
```suggestion
            <DefaultSwitchControl
              name="emailVerified"
              label={t("emailVerified")}
              labelIcon={t("emailVerifiedHelp")}
            />
```

Reply:
No the default has the labels "off" and "on" here we want "yes" and "no"

Reply:
Ah my bad, consider this resolved.